s5_subwords/readme.md

```diff
@@ -1,19 +1,50 @@
-Work in progress
-
-
 # Subword recipe for Icelandic
-This is a subword-based recipe for an Icelandic ASR. Running the recipe on other languages only requires modifications to the dataset-specific data preparation steps. Subword-based ASR models have the advantage of being able to create OOV words. 
+This is a subword-based recipe for an Icelandic ASR build using the Kaldi toolkit. Running the recipe on other languages only requires modifications to the dataset-specific data preparation steps. Subword-based ASR models have the advantage of being able to create out-of-vocabulary words. 
 
 ## Running the recipe 
-In the folder "run" are all the scripts needed to run the recipe. The acoustic and subword modelling are in separate scripts. The subword related modifications are only done to the lexicon and language model except that the phonetic units that the acoustic model creates are graphemes instead of phonemes. 
+In the folder "run" are the scripts used to interface the recipe. The acoustic and subword modelling are in separate scripts. The subword related modifications are only done to the lexicon and language model, except that the phonetic units used for the acoustic model are graphemes instead of phonemes. 
+
+### run/run_am.sh
+Start by running the script "run/run_am.sh", it has all the steps related to the acoustic model training. It creates a standard LDA+MLLT+SAT system followed by a TDNN model, with i-vectors and speed perturbation. The script is a standard Kaldi recipe except for the creation of the dictionary as it uses the text corpus to create a grapheme based lexicon instead of a phoneme based model. The "lang" and "dict" used for the acoustic model training are prefixed with "base" to distinguish them from their subword counterparts, which will be created later. The "lang" and "dict" are derived from the text corpus, so no g2p translation is required. Note that the text corpus must be normalized; one sentence per line, all characters should be lower or upper case, and all non-character related signs like numbers or punctuations should be removed. 
+
+### Subword model
+The subword realted steps are in run/run_sw.sh. There are scripts for three subword tokenizers:
+ * [Unigram](https://arxiv.org/abs/1804.10959) 
+ * [Byte Pair Encoding](https://arxiv.org/abs/1508.07909)
+ * [Morfessor](https://morfessor.readthedocs.io/en/latest/)
+ 
+Please read about the tokenizer in their respective repositories, they all perform the same task, but the WER results will differ for each method. The default tokenizer is a "bpe", all scripts for that tokenizer are provided with this repository. To run the tokenizers "unigram" and the "sp_bpe" [Google's Sentence Piece](https://github.com/google/sentencepiece) is needed, a code snippet for a python virtual environment is provided along with a requirments.txt file. 
```

Comment:
1. what do you mean by specific repositories? The links are only to their papers, except for the Morfessor one.
2. Can you add Google's Sentence Piece repo as a submodule?
```suggestion
Please read about the tokenizer in their respective repositories, they all perform the same task, but the WER results will differ for each method. Our default tokenizer is "bpe", all scripts for that tokenizer are provided with this repository. To run the tokenizers "unigram" and the "sp_bpe," [Google's Sentence Piece](https://github.com/google/sentencepiece) is needed, a code snippet for a python virtual environment is provided along with a requirements.txt file. 
```